### ISSUE TITLE
@cast on functions with tuples as lvalues

### ISSUE THREAD

**Issue by prittjam** [mcabbott/TensorCast.jl#61]
is there a way to use tensor cast for functions that return tuples? e.g., 
```
function axis_to_random_rotation3(x::AbstractVector)
    x′ = similar(x)
    x′ .= rand(eltype(x),3)
    y = cross(x,x′)
    z = cross(x,y)
    R = [x y z]
    R .= R./norm.(eachcol(R))'

    return R
end

renorm!(n) = n ./= norm(n)

@views function plane_to_random_pose(Π)
    renorm!(Π)
    R = axis_to_random_rotation3(Π[1:3])
    t = -Π[4].*Π[1:3]
    
    return R, t
end

```

Here I want to use tensor cast to shape the returned pose from a set of planes e.g. 
````
Π = rand(4,100)
@cast something something  := plane_to_random_pose(Π[:,j])[i]
````

**Comment by mcabbott** [comment]
This is essentially https://github.com/mcabbott/TensorCast.jl/issues/15 . It would be nice to support, using StructArrays, but I haven't got around to writing it.

**Comment by prittjam** [comment]
Thanks for the response.  I think unwrapping directly to arrays from a broadcasted function with multiple return values should be fixed at the language level. Tensors are a common data representation, and broadcasting on them is awkward except for the simplest case with mapslices. @cast addresses some issues albeit the return types can be very complicated, but there are still common use cases that are not handled elegantly. There are some threads about this topic, so I don't think I'm the only one feeling the pain. Given that array is a fundamental Julia datatype, this lack of functionality surprises me.  I also suspect I'm paying some hidden performance penalities with casting. I'm not yet a savvy enough Julia user to understand the tradeoffs.

**Comment by mcabbott** [comment]
Yes, lots of people would like this, I mean outside of TensorCast. 

StructArrays seems to provide a pretty good implementation of this `unzip_broadcast` idea. I doubt anyone wants to build that into Base, but pulling out just enough to do this might be possible, haven't looked. 

`mapslices`-like things we are trying to improve with things like `stack(f, eachcol(x), y)`. Maybe there is more to be done at the intersection of these two.
